src/server/string_family.cc

```diff
@@ -832,86 +856,90 @@ void StringFamily::SetNx(CmdArgList args, ConnectionContext* cntx) {
   return builder->SendLong(0);  // value do exists, we need to report that we didn't change it
 }
 
+// With tieringV2 support
 void StringFamily::Get(CmdArgList args, ConnectionContext* cntx) {
   auto cb = [key = ArgS(args, 0)](Transaction* tx, EngineShard* es) -> OpResult<StringValue> {
     auto it_res = es->db_slice().FindReadOnly(tx->GetDbContext(), key, OBJ_STRING);
     if (!it_res.ok())
       return it_res.status();
 
-    if (const PrimeValue& pv = (*it_res)->second; pv.IsExternal()) {
-      return {es->tiered_storage_v2()->Read(key, pv)};
-    } else {
-      std::string buf;
-      pv.GetString(&buf);
-      return {std::move(buf)};
-    }
+    return StringValue::Read(key, (*it_res)->second, es);
   };
 
-  auto res = cntx->transaction->ScheduleSingleHopT(cb);
-
-  auto* rb = static_cast<RedisReplyBuilder*>(cntx->reply_builder());
-  switch (res.status()) {
-    case OpStatus::OK:
-      rb->SendBulkString(std::move(res.value()).Get());
-      break;
-    case OpStatus::WRONG_TYPE:
-      rb->SendError(kWrongTypeErr);
-      break;
-    default:
-      rb->SendNull();
-  }
+  StringReplies{cntx->reply_builder()}.Send(cntx->transaction->ScheduleSingleHopT(cb));
 }
 
+// With tieringV2 support
 void StringFamily::GetDel(CmdArgList args, ConnectionContext* cntx) {
-  string_view key = ArgS(args, 0);
+  auto cb = [key = ArgS(args, 0)](Transaction* tx, EngineShard* es) -> OpResult<StringValue> {
+    auto it_res = es->db_slice().FindMutable(tx->GetDbContext(), key, OBJ_STRING);
+    if (!it_res.ok())
+      return it_res.status();
 
-  auto cb = [&](Transaction* t, EngineShard* shard) {
-    bool run_del = true;
-    return OpMutableGet(t->GetOpArgs(shard), key, run_del);
```

Comment:
seems that you can delete OpMutableGet ?

Reply:
It's still used from GetEx, I'll do that in the next PR